wen-restart/src/wen_restart.rs

```diff
@@ -737,7 +754,120 @@ pub(crate) fn aggregate_restart_heaviest_fork(
         if time_left > 0 {
             sleep(Duration::from_millis(time_left));
         }
+        // Print the block stake map after a while.
+        if stat_printed_at.elapsed() > Duration::from_secs(COORDINATOR_STAT_PRINT_INTERVAL_SECONDS)
+        {
+            heaviest_fork_aggregate.print_block_stake_map();
+            stat_printed_at = Instant::now();
+        }
+    }
+}
+
+pub(crate) fn repair_heaviest_fork(
+    my_heaviest_fork_slot: Slot,
+    heaviest_slot: Slot,
+    exit: Arc<AtomicBool>,
+    blockstore: Arc<Blockstore>,
+    wen_restart_repair_slots: Arc<RwLock<Vec<Slot>>>,
+) -> Result<()> {
+    loop {
+        if exit.load(Ordering::Relaxed) {
+            return Err(WenRestartError::Exiting.into());
+        }
+        let to_repair = if blockstore.meta(heaviest_slot).is_ok_and(|x| x.is_some()) {
```

Comment:
It's possible the `heaviest_slot` is not included in `wen_restart_repair_slots`, this loop repairs `heaviest_slot` and its ancestors greater than `my_heaviest_fork_slot`. If heaviest_slot is already full, it could be filtered out on line
`.filter(|slot| !blockstore.is_full(*slot))`, then it's not in repair list.

If `heaviest_slot` is a valid block, we should be able to repair its ancestors until eventually we hit an ancestor smaller than `my_heaviest_fork_slot`, so the loop should terminate.

I added comments, does this look clearer?

Reply:
makes sense 👍